### PR TITLE
Install deps via requirements.txt

### PR DESCRIPTION
**Pull request by inducer**

### CODE REVIEW

**Comment by matthiasdiener** [comment]
pip doesnt like building pyopencl against conda's pocl for some reason:

```
gcc -pthread -B /usr/share/miniconda/envs/test/compiler_compat -Wl,--sysroot=/ -Wsign-compare -fwrapv -Wall -O3 -DNDEBUG -fPIC -DPYGPU_PACKAGE=pyopencl -DPYGPU_PYOPENCL=1 -Ipybind11/include -I/tmp/pip-build-env-hjzvlyxy/overlay/lib/python3.8/site-packages/pybind11/include -I/tmp/pip-build-env-hjzvlyxy/overlay/lib/python3.8/site-packages/pybind11/include -I/tmp/pip-build-env-hjzvlyxy/overlay/lib/python3.8/site-packages/numpy/core/include -I/usr/share/miniconda/envs/test/include/python3.8 -c src/wrap_constants.cpp -o build/temp.linux-x86_64-3.8/src/wrap_constants.o -fvisibility=hidden -DVERSION_INFO="2020.2" -std=gnu++14 -fvisibility=hidden
  In file included from src/wrap_constants.cpp:30:0:
  src/wrap_cl.hpp:57:10: fatal error: CL/cl.h: No such file or directory
   #include <CL/cl.h>
            ^~~~~~~~~
  compilation terminated.
  error: command 'gcc' failed with exit status 1
```

**Comment by matthiasdiener** [comment]
It gets further now, current error is:

```
=================================== FAILURES ===================================
_ test_wave_accuracy[<context factory for <pyopencl.Device 'pthread-Intel(R) Xeon(R) Platinum 8171M CPU @ 2.60GHz' on 'Portable Computing Language'>-4-problem3] _
Traceback (most recent call last):
  File "/home/runner/work/mirgecom/mirgecom/.miniforge3/envs/testing/lib/python3.8/site-packages/pymbolic/mapper/__init__.py", line 126, in __call__
    method = getattr(self, expr.mapper_method)
AttributeError: 'numpy.ndarray' object has no attribute 'mapper_method'
```

**Comment by inducer** [comment]
The actual error was further below:
```
pyopencl._cl.MemoryError: create_buffer failed: MEM_OBJECT_ALLOCATION_FAILURE
```
It was simply out of memory. In 0a0b4bc, I've reduced the problem scale in 3D.

cc @majosm

File: test/test_wave.py [hunk 1]
```diff
@@ -170,7 +170,8 @@ def sym_eval(expr, t):
 
         expected_rhs = sym_eval(sym_rhs, t_check)
 
-        eoc_rec.add_data_point(1./n, max_inf_norm(rhs - expected_rhs))
+        rel_linf_err = max_inf_norm(rhs - expected_rhs)/max_inf_norm(expected_rhs)
```
Comment (inducer) [inline test/test_wave.py:173]: FYI @majosm

**Pull request merged**